ibm/resource_ibm_cloudant_test.go

```diff
@@ -0,0 +1,217 @@
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckIBMCloudantExists(resourceName, conf),
+					resource.TestCheckResourceAttr(resourceName, "name", serviceName),
+					resource.TestCheckResourceAttr(resourceName, "service", "cloudantnosqldb"),
+					resource.TestCheckResourceAttr(resourceName, "plan", "lite"),
+					resource.TestCheckResourceAttr(resourceName, "legacy_credentials", "false"),
+					resource.TestCheckResourceAttr(resourceName, "capacity", "1"),
+					resource.TestCheckResourceAttr(resourceName, "throughput.read", "20"),
+					resource.TestCheckResourceAttr(resourceName, "enable_cors", "true"),
+					resource.TestCheckResourceAttr(resourceName, "cors_config.0.allow_credentials", "true"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"parameters"},
+			},
+		},
+	})
+}
+
+func testAccCheckIBMCloudantDestroy(s *terraform.State) error {
+	rsContClient, err := testAccProvider.Meta().(ClientSession).ResourceControllerAPI()
+	if err != nil {
+		return err
+	}
+	for rsName, rs := range s.RootModule().Resources {
+		if rs.Type != "ibm_cloudant" || !strings.HasPrefix(rsName, "terraform-test-") {
+			continue
+		}
+
+		instanceID := rs.Primary.ID
+
+		// Try to find the key
+		instance, err := rsContClient.ResourceServiceInstance().GetInstance(instanceID)
+
+		if err == nil {
+			if !reflect.DeepEqual(instance, models.ServiceInstance{}) && instance.State == "active" {
+				return fmt.Errorf("Resource Instance still exists: %s", rs.Primary.ID)
+			}
+		} else {
+			if !strings.Contains(err.Error(), "404") {
+				return fmt.Errorf("Error checking if Resource Instance (%s) has been destroyed: %s", rs.Primary.ID, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckIBMCloudantExists(resourceName string, obj models.ServiceInstance) resource.TestCheckFunc {
+
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		rsContClient, err := testAccProvider.Meta().(ClientSession).ResourceControllerAPI()
+		if err != nil {
+			return err
+		}
+		instanceID := rs.Primary.ID
+
+		instance, err := rsContClient.ResourceServiceInstance().GetInstance(instanceID)
+
+		if err != nil {
+			return err
+		}
+
+		obj = instance
+		return nil
+	}
+}
+
+func testAccCheckIBMCloudantResourceConfig(serviceName string) string {
+	return fmt.Sprintf(`
+
+	resource "ibm_cloudant" "instance" {
+		name                = "%s"
+		plan                = "standard"
+		location            = "us-south"
+
+		legacy_credentials  = true
+		include_data_events = true
+		capacity            = 1
+		enable_cors         = true
+
+		cors_config {
+			allow_credentials = false
+			origins           = ["https://example.com"]
+		}
+
+		timeouts {
+		  create = "15m"
+		  update = "15m"
+		  delete = "15m"
+		}
```

Comment:
I'd like a little clarification on this.
My understanding from [Terraform docs](https://www.terraform.io/docs/language/resources/syntax.html#operation-timeouts) is that this is a convention, but entirely handled by the provider, but I think they mean the implementation of the timeout because the [Resource](https://www.terraform.io/docs/extend/resources/retries-and-customizable-timeouts.html) struct appears to have customizable timeout properties.
The [IBM provider docs](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/resource_instance#timeouts) talk about a 10 minute default timeout (for `ibm_resource_instance`) and I can see that we have configured a [hard-coded 10 minute](https://github.com/cloudant/terraform-provider-ibm/blob/27393bb2232dabb7828131baa6492f88270f3b50/ibm/resource_ibm_cloudant.go#L146) default timeout.

AFAICT the timeout is divided into two parts `ResourceTimeout` and `DefaultTimeout` and config blobs like this should be setting the former and the hard-coding in our provider is setting the latter. Can you confirm if that is correct/also your understanding? I couldn't find the relevant code that actually differentiates these two things.

Reply:
Right, so those are two sides of the same medal. Internally CRUD `Timeouts` set during definition of `schema.Resource` go to `schema.ResourceData` private attribute `timeouts` and this conventional config's "timeouts" block overwrites it. Implementation of timeout left to provider* and timeout's value can be accessed by `Timeout(key)` function with default [hardcoded 20 mins](https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/ibm/resource_ibm_resource_instance.go#L796). 

We are setting `Timeouts` to 10 mins on our `schema.Resource` to match `ibm_resource_instance`, but not using them directly, instead relying on [ibm_resource_instance implementation](https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/ibm/resource_ibm_resource_instance.go#L796) for CRUD operations.

Timeouts overwritten here to match resource instance test. Though I never seen acceptance test stages to take longer than 8 mins, I set it here just in case upstream's testing infra is busier and actually takes longer to provision instances.

* - To be honest I'm not even sure if terraform sdk enforces the timeout if provider hasn't implemented it or it'll hung on none-returning function forever.

Reply:
OK thanks